tests/numerics/fv/test_tpsa.py

```diff
@@ -21,1418 +36,1426 @@
+        sps.sparray: Divergence matrix for the face terms.
+        sps.sparray: Accumulation matrix for the cell center terms.
 
-        # The contribution from cell center displacement to the boundary displacement.
-        bound_displacement_cell = np.zeros((4, 4))
-        bound_displacement_cell[0, 0] = c2f_avg_0_x
-        bound_displacement_cell[1, 1] = c2f_avg_0_y
-        bound_displacement_cell[2, 2] = c2f_avg_6
-        bound_displacement_cell[3, 3] = c2f_avg_6
+    """
+    C = d[pp.PARAMETERS][KEYWORD]["fourth_order_tensor"]
 
-        # Prescribed stresses are converted to displacements by 'inverting' Hook's law.
-        # Multiply with -1 on face 0 since this has an inward pointing normal vector.
-        bound_displacement_face = np.zeros((4, 14))
-        bound_displacement_face[0, 0] = -c2f_avg_0_x_bound / (
-            (2 * mu_0_d + rw_0_x) * self.n_0_nrm
-        )
-        bound_displacement_face[1, 1] = -c2f_avg_0_y_bound / (
-            (2 * mu_0_d + rw_0_y) * self.n_0_nrm
-        )
-        bound_displacement_face[2, 12] = c2f_avg_6_bound / (
-            (2 * mu_1_d + rw_6) * self.n_6_nrm
-        )
-        bound_displacement_face[3, 13] = c2f_avg_6_bound / (
-            (2 * mu_1_d + rw_6) * self.n_6_nrm
-        )
+    # Deal with the different dimensions of the rotation variable.
+    rot_dim = g.dim if g.dim == 3 else 1
 
-        bound_displacement_rotation_cell = np.zeros((4, 2))
-        bound_displacement_rotation_cell[0, 0] = (
-            self.n_0[1] * c_c2f_avg_0_x / ((2 * mu_0_d + rw_0_x) * self.n_0_nrm)
-        )
-        bound_displacement_rotation_cell[1, 0] = (
-            -self.n_0[0] * c_c2f_avg_0_y / ((2 * mu_0_d + rw_0_y) * self.n_0_nrm)
-        )
-        bound_displacement_rotation_cell[2, 1] = (
-            -self.n_6[1] * c_c2f_avg_6 / ((2 * mu_1_d + rw_6) * self.n_6_nrm)
-        )
-        bound_displacement_rotation_cell[3, 1] = (
-            self.n_6[0] * c_c2f_avg_6 / ((2 * mu_1_d + rw_6) * self.n_6_nrm)
-        )
+    n_rot_face = g.num_faces * rot_dim
+    n_rot_cell = g.num_cells * rot_dim
+    div_rot = g.divergence(dim=rot_dim)
 
-        # Contribution from solid pressure. Multiply with -1 on face 0 since this has an
-        # inward pointing normal vector.
-        bound_displacement_solid_pressure_cell = np.zeros((4, 2))
-        bound_displacement_solid_pressure_cell[0, 0] = (
-            -self.n_0[0] * c_c2f_avg_0_x / ((2 * mu_0_d + rw_0_x) * self.n_0_nrm)
-        )
-        bound_displacement_solid_pressure_cell[1, 0] = (
-            -self.n_0[1] * c_c2f_avg_0_y / ((2 * mu_0_d + rw_0_y) * self.n_0_nrm)
-        )
-        bound_displacement_solid_pressure_cell[2, 1] = (
-            self.n_6[0] * c_c2f_avg_6 / ((2 * mu_1_d + rw_6) * self.n_6_nrm)
-        )
-        bound_displacement_solid_pressure_cell[3, 1] = (
-            self.n_6[1] * c_c2f_avg_6 / ((2 * mu_1_d + rw_6) * self.n_6_nrm)
-        )
+    flux = sps.block_array(
```

Comment:
Why do you call this "flux"?

Reply:
Habit: First block row is stress, second is face rotations, third is actually flux of solid mass. Scalings that alter the physical interpretation may apply.

Renamed to face_quantities for lack of a better expression.